scripts/clean_osm_data.py

```diff
@@ -852,7 +911,7 @@ def clean_data(
 
     if os.path.getsize(input_files["lines"]) > 0:
         # Load raw data lines
-        df_lines = gpd.read_file(input_files["lines"])
+        df_lines = use_custom_data_files("lines")
```

Comment:
If you like the proposal above with the definition like `load_network_data(network_asset, clean_options, osm_paths)`, then we need to these function calls can be revised as:
`load_network_data("lines", clean_options, input_files)`

where clean_option shall be added as argument to the function clean_data and called with value `snakemake.params.clean_osm_data_options`.

By passing the entire `clean_options` dictionary, we could clean the arguments of the function clean_data, but this may be optional

Reply:
Hey @davide-f ! Thanks for all the comments, they were super useful to make the PR more concise. Some comments on this update:
- I adopted the "try" command to simplify the ifs and elifs in the code (which did cut down a bunch of lines and made everything nicer to read)
- I also used the dynamic calling of the "f-string" which simplified the number of variables needed to be defined so, at the end, the network_asset was the only argument needed (aside from the inputs from the config file)
Let me know how it looks and if it needs any additional changes!